<?xml version="1.0"?>
<launch>
    <arg name="vehicle" default="iris_rplidar" />
    <arg name="world_name" default="empty_room" />
    <arg name="pos_x" default="0" />
    <arg name="pos_y" default="0" />
    <arg name="pos_z" default="0" />
    <include file="$(find px4)/launch/mavros_posix_sitl.launch">
	<!-- MAVROS posix SITL environment launch script -->
    	<!-- launches MAVROS, PX4 SITL, Gazebo environment, and spawns vehicle -->
    	<!-- vehicle pose -->
    	<arg name="x" value="$(arg pos_x)"/>
    	<arg name="y" value="$(arg pos_y)"/>
    	<arg name="z" value="$(arg pos_z)"/>
    	<arg name="R" value="0"/>
    	<arg name="P" value="0"/>
    	<arg name="Y" value="0"/>
    	<!-- vehicle model and world -->
    	<arg name="est" value="ekf2"/>
    	<arg name="world" value="$(find indoor_navigation)/worlds/$(arg world_name).world"/>
	<arg name="sdf" value="$(find indoor_navigation)/models/$(arg vehicle)/$(arg vehicle).sdf"/>

    	<!-- gazebo configs -->
    	<arg name="gui" value="true"/>
    	<arg name="debug" value="false"/>
    	<arg name="verbose" value="false"/>
    	<arg name="paused" value="false"/>
    	<arg name="respawn_gazebo" value="false"/>
    	<!-- MAVROS configs -->
    	<arg name="fcu_url" value="udp://:14540@localhost:14557"/>
    	<arg name="respawn_mavros" value="false"/>
    	<!-- PX4 configs -->
    	<arg name="interactive" value="true"/>
    </include>

    <!-- offboard mode and arm vehicle -->
    <node name="switch_mode_node" pkg="indoor_navigation" type="switch_mode.py" output="screen"/>

    <param name="/px4/gazebo_world_name" type="str" value="$(arg world_name)"/>
    <param name="/px4/vehicle_type" type="str" value="$(arg vehicle)"/>
    <rosparam file="$(find indoor_navigation)/config/config.yaml" command="load" ns="/px4"/>
</launch>
